File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
int i = 0;
bool comp(string s1, string s2)
{
    if (s1[i] == s2[i])
        return s1 < s2;
    else
        return s1[i] < s2[i];
}
vector<string> solution(vector<string> strings, int n)
{
    i = n;
    sort(strings.begin(), strings.end(), comp);
    return strings;
}

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String[] solution(String[] strings, int n) {
        Arrays.sort(strings,
                Comparator.comparing((String str) -> str.charAt(n))
                        .thenComparing(str -> str)
        );
        return strings;
    }
}